.spin/cmds.py

```diff
@@ -33,37 +42,12 @@ def docs(ctx, clean, install_deps, build):
       SPHINXOPTS="" spin docs
 
     """
-    if clean:
-        doc_dirs = [
-            "./doc/build/",
-            "./doc/source/api/",
-            "./doc/source/auto_examples/",
-            "./doc/source/jupyterlite_contents/",
-        ]
-        for doc_dir in doc_dirs:
-            if os.path.isdir(doc_dir):
-                print(f"Removing {doc_dir!r}")
-                shutil.rmtree(doc_dir)
-
-    if build:
-        click.secho(
-            "Invoking `build` prior to running tests:", bold=True, fg="bright_green"
-        )
-        ctx.invoke(meson.build)
-
-    try:
-        site_path = meson._get_site_packages()
-    except FileNotFoundError:
-        print("No built scikit-image found; run `spin build` first.")
-        sys.exit(1)
-
     if install_deps:
         util.run(['pip', 'install', '-q', '-r', 'requirements/docs.txt'])
 
-    os.environ['SPHINXOPTS'] = os.environ.get('SPHINXOPTS', "-W")
-
-    os.environ['PYTHONPATH'] = f'{site_path}{os.sep}:{os.environ.get("PYTHONPATH", "")}'
-    util.run(['make', '-C', 'doc', 'html'], replace=True)
+    for extra_param in ('install_deps',):
+        del ctx.params[extra_param]
```

Comment:
Is spin always adding the param `install_deps`? I don't really understand what this does.

Reply:
Ah, because `click.pass_context` includes the parsed parameters as well.